code/modules/mob/living/bloodcrawl.dm

```diff
@@ -0,0 +1,109 @@
+//Travel through pools of blood. Slaughter Demon powers for everyone!
+
+/mob/living/proc/phaseout(obj/effect/decal/cleanable/B)
+	var/mob/living/kidnapped = null
+	var/turf/mobloc = get_turf(src.loc)
+	var/turf/bloodloc = get_turf(B.loc)
+	if(Adjacent(bloodloc))
+		src.notransform = TRUE
+		spawn(0)
+			src.visible_message("[src] sinks into the pool of blood.")
+			playsound(get_turf(src), 'sound/magic/enter_blood.ogg', 100, 1, -1)
+			var/obj/effect/dummy/slaughter/holder = new /obj/effect/dummy/slaughter( mobloc )
```

Comment:
Well, swap new for PoolOrNew(), look up the doc on that proc
and swap qdel for ReturnToPool() (OR, and this is probably better, override Destroy() to return QDEL_HINT_PLACE_IN_POOL)

a good example would probably be fire hotspots, or virtual speakers.